pkg/scheduler/framework/runtime/framework.go

```diff
@@ -1226,7 +1226,6 @@ func (f *frameworkImpl) WaitOnPermit(ctx context.Context, pod *v1.Pod) *framewor
 	if !s.IsSuccess() {
 		if s.IsUnschedulable() {
 			klog.V(4).InfoS("Pod rejected while waiting on permit", "pod", klog.KObj(pod), "status", s.Message())
-			s.SetFailedPlugin(s.FailedPlugin())
```

Comment:
Better to remain this line and only return status in `Reject`. It's more readability. Or we have other reasons blocking this.

Reply:
I may get the point. It probably looks like this?
```go
// WaitOrPermit
if s.IsUnschedulable() {
	klog.V(4).InfoS("Pod rejected while waiting on permit", "pod", klog.KObj(pod), "status", s.Message())
	return s.WithFailedPlugin("xxxx")
}
// Reject
select {
      case w.s <- framework.NewStatus(framework.Unschedulable, msg):
      default:
}
```

But I don't know how to get the `FailedPlugin` in `WaitOrPermit`.

Reply:
hmm.. I misunderstood the intension here, forget about that. It looks great.